server/api/mixed-content/index.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * @fileoverview API Route
+ * Route for mixed content queries, it uses BigQuery's
+ * client and express's router.
+ *
+ * Queries are stored in /.queries.json, each has a name, description,
+ * sqlQuery and suggested visualizations.
+ *
+ */
+
+import {Router as router} from 'express';
+import {BigQuery} from '@google-cloud/bigquery';
+// This is a collection of all queries and their metadata in json.
+import * as queries from './queries.json';
+
+const mixedApi = router();
+const bigqueryClient = new BigQuery();
+
+mixedApi.get('/top-websites-with-mixed-content', async (req, res) => {
+  const query = await queries.TopWebsitesWithMixedContent;
+  const rows = await queryData(query);
+
+  res.json({
+    description: query.description,
+    result: rows,
+    suggestedVisualizations: query.suggestedVisualizations,
+  });
+});
+
+mixedApi.get(
+    '/top-government-websites-with-mixed-content',
+    async (req, res) => {
+      const query = await queries.TopGovernmentWebsitesWithMixedContent;
+      let rows = [];
+      rows = await queryData(query);
+
+      res.json({
+        description: query.description,
+        result: rows,
+        suggestedVisualizations: query.suggestedVisualizations,
+      });
+    },
+);
+
+mixedApi.get('/top-countries-with-more-government-websites-with-mixed-content',
+    async (req, res) => {
+      const query = await
+      queries.TopCountriesWithMoreGovernmentWebsitesWithMixedContent;
+      const rows = await queryData(query);
+
+      res.json({
+        description: query.description,
+        result: rows,
+        suggestedVisualizations: query.suggestedVisualizations,
+      });
+    },
+);
+
+mixedApi.get('/mixed-content-percentage-histogram', async (req, res) =>{
+  const query = await queries.MixedContentPercentageHistogram;
+  let rows = [];
+  rows = await queryData(query);
+
+  res.json({
+    description: query.description,
+    result: rows,
+    suggestedVisualizations: query.suggestedVisualizations,
+  });
+});
+
+
+/**
+ * Makes a BigQuery query given the query from ./queries.json
+ * @param {object} data Query from /.queries.json
+ * @return {object} Array of rows (result of the query).
+ */
+const queryData = async (data) => {
+  if (typeof data.query === 'object') {
```

Comment:
Is there a case where the query is not an array? I think we can specify its type in the top comment.

Also, if the query is always going to be an array, I recommend you to use `Array.isArray()`, you can find docs [here](https://developer.mozilla.org/es/docs/Web/JavaScript/Referencia/Objetos_globales/Array/isArray)

Reply:
Sometimes it gets passed as an array and other times it's just a string, I really don't know why that happens. That was a quick solution I could think of.

Reply:
Should we leave it like that? (taking into consideration Array.isArray())

Reply:
Got it.

I think is okay since the input depends totally on us. But if we accidentally add something that is an object `{}`, it'll pass the if statement and will throw an error like `.join(' ') is not a method of data.query`.

I recommend the following,

1.- Destructure the param `data` into `async ({query})`
2.- Update the comment, addressing that `query` should be an array or a string.

That would make it clear